tabix.c

```diff
@@ -49,15 +50,34 @@ typedef struct
 }
 args_t;
 
-static void error(const char *format, ...)
+HTS_FORMAT(HTS_PRINTF_FMT, 1, 2) static void error(const char *format, ...)
 {
     va_list ap;
+    fflush(stdout);
     va_start(ap, format);
     vfprintf(stderr, format, ap);
     va_end(ap);
+    fflush(stderr);
     exit(EXIT_FAILURE);
 }
 
+HTS_FORMAT(HTS_PRINTF_FMT, 1, 2) static void error_errno(const char *format, ...)
+{
+    va_list ap;
+    fflush(stdout);
+    if (format) {
+        va_start(ap, format);
+        vfprintf(stderr, format, ap);
+        va_end(ap);
+    }
+    if (errno) {
+        fprintf(stderr, "%s%s\n", format ? ": " : "", strerror(errno));
```

Comment:
`fflush` and `vfprintf` might clobber `errno`, so best to sample it at the top of the function.

Reply:
Good point, I now take a copy.